### DISCUSSION TITLE
レイヤー数の上限について

### DISCUSSION THREAD

**Discussion by fal-math** [Yowkees/keyball#118] [Q&A] [answered]
お世話になっております.
先日keyball39を組み立てサービスを利用して購入したものです.（正常に動作しております、ありがとうございます.）

keyball (初代)を以前より常用しております、どちらもレイヤー数がデフォルトだと4つまでになっていますが、ProMicro の制約上（メモリ上限？）そうなっているのでしょうか.それとも、ファームウェアの書き換え等を行えばレイヤー数を増やせますでしょうか.（ファームウェアの書き換え経験はないので、具体的にどうすれば良いのかはわかっておりません...）

個人的にはkeyball, keyball39ともに、あと1レイヤーあると嬉しいと思っております.

お忙しいところ恐縮ですがご回答いただければ幸いです.

**Comment by Yowkees** [comment, marked as answer]
Keyball39のご購入誠にありがとうございます！
Via(Remap)の制約上、レイヤー数上限は４つとなっていると思っています。
ただ、ファームウェアの書き換えが可能でしたらレイヤー数は増やせます。（その際、Remapは使用できなくなるかもです）

https://salicylic-acid3.hatenablog.com/entry/via-support
こちらの記事でレイヤー数の書き換えなどの方法が載っていましたのでご参考になればと。
メモリ容量がオーバーするようであれば、LEDのアクションをいくつかOFFにしていただければと思います。
この程度のご回答で申し訳ありませんがご検討お願いいたします。

**Comment by fal-math** [comment]
ご回答ありがとうございます！

ファームウェアの書き換え、時間があるときにチャレンジしてみようと思います。参考記事もありがとうございます。

Remapは、うまくやったら使えるのかも、とは思っています。実際、4レイヤーより多く表示されているキーボードもありました。
(例として "Jones" の "wide us and num layer" を貼っておきます。
https://remap-keys.app/catalog/z8HynboRCOQfPmwXbzp6/keymap?id=NXDzVJVuxeW8HdR5joy8 )
ただ、実際やってみないと分からないですね。今度試してみます。

**Reply by Yowkees** [reply]
4レイヤー以上のキーボードもあるのですね！
勉強不足でした。

Keyball39はキー数少ないのでそのような要望ありそうです。
レイヤーひとつ増やせるように私も時間ができた際に試してみます。
ご指摘ありがとうございましたm(_ _)m
